File: pokemon-battle.py
class pikachu():
	hp = 100
	ataque = 55

	def win(self):
		print "pikachu win"

class jigglypuff():
	hp = 100
	ataque = 45

	def win(self):
		print "Jigglypuff win"

turno = 1

while(pikachu.vida > 0 and jigglypuff.vida > 0):
	if (turno == 1):
		jigglypuff.vida = jigglypuff.vida - pikachu.ataque
	else:
		pikachu.vida = pikachu.vida - jigglypuff.ataque

if(pikachu.hp<=0):
	jigglypuff.win()
else:
	pikachu.win()

File: pokemon-battle.js
var Pikachu = {
	"hp":100,
	"ad":55,
	"win":function(){
		console.log("Pikachu Win!");
	}
};

var Jigglypuff = {
	"hp":100,
	"ad":45,
	"win":function(){
		console.log("Jigglypuff Win!");
	}
};

turno = Math.floor(Math.random()*2);

while(Pikachu.hp > 0 && Jigglypuff.hp > 0){
	if (turno == 1){
		Jigglypuff.hp = Jigglypuff.hp - Pikachu.ad;
		console.log("Ataque de Pikachu");
		turno = 0;
	}else{
		Pikachu.hp = Pikachu.hp - Jigglypuff.ad;
		console.log("Ataque de Jigglypuff");
		turno = 1;
	}
}

if(Pikachu.hp <= 0){
	Jigglypuff.win()
}else{
	Pikachu.win()
}